lib/src/repo.rs

```diff
@@ -1353,6 +1353,22 @@ impl MutableRepo {
         Ok(())
     }
 
+    pub fn rename_workspace(
+        &mut self,
+        old_workspace_id: &WorkspaceId,
+        new_workspace_id: WorkspaceId,
+    ) -> Result<(), RenameWorkspaceError> {
+        let wc_commit_id = self
+            .view_mut()
+            .remove_wc_commit(old_workspace_id)
```

Comment:
nit: maybe better to add `view.rename_workspace()` instead of making `.remove_wc_commit()` return old data?

Reply:
Sounds good. Done. (also moved `RenameWorkspaceError` to view.rs)